WordPress/Classes/Services/PostCoordinator.swift

```diff
@@ -365,6 +365,10 @@ class PostCoordinator: NSObject {
         wpAssert(isSyncAllowed(for: revision.original()), "Sync is not supported for this post")
 
         if !revision.isSyncNeeded {
+            // Set the foreignID if the revision doesn't have a postID or foreignID assigned yet.
+            if let postID = revision.postID?.intValue, postID <= 0, revision.foreignID == nil {
+                revision.foreignID = UUID().uuidString
```

Comment:
The best place to assign a `foregnID` is right during the post created (see [`createPost`](https://github.com/wordpress-mobile/WordPress-iOS/blob/7386d4ebee889c87a62e79208662e42f8fcc83fe/WordPress/Classes/Models/Blog%2BPost.swift#L40) so that every post object has one.

(Nit) if you use a `UUID` type in Core Data, it'll take less space. Binary `UUID` takes 16 bytes. The string representation is 36 bytes.

Reply:
Fixed! [13d513a](https://github.com/wordpress-mobile/WordPress-iOS/pull/23173/commits/13d513a9cd5c905508ac9acc280e32487f545fc4), [92a9488](https://github.com/wordpress-mobile/WordPress-iOS/pull/23173/commits/92a94883b27c393c29db62f4e915da9663c4a868)